packages/angular/src/data-connect/index.ts

```diff
@@ -0,0 +1,289 @@
+import {
+  type CreateMutationOptions,
+  type CreateMutationResult,
+  type CreateQueryOptions,
+  type CreateQueryResult,
+  injectMutation,
+  injectQuery,
+  QueryClient,
+  type QueryKey,
+} from "@tanstack/angular-query-experimental";
+
+import type { FirebaseError } from "firebase/app";
+
+import { inject, Injector, signal } from "@angular/core";
+import {
+  type CallerSdkType,
+  CallerSdkTypeEnum,
+  DataConnect,
+  executeMutation,
+  executeQuery,
+  type MutationRef,
+  type MutationResult,
+  type QueryRef,
+  type QueryResult,
+} from "@angular/fire/data-connect";
+function getQueryKey(queryRef: QueryRef<unknown, unknown>) {
+  const key: (string | Record<string, any>)[] = [queryRef.name];
+  if ("variables" in queryRef && queryRef.variables !== undefined) {
+    key.push(queryRef.variables as unknown as Record<string, any>);
+  }
+  return key;
+}
+export type FlattenedQueryResult<Data, Variables> = Omit<
+  QueryResult<Data, Variables>,
+  "data" | "toJSON"
+> &
+  Data;
+export interface CreateDataConnectQueryOptions<Data, Variables>
+  extends Omit<
+    CreateQueryOptions<
+      FlattenedQueryResult<Data, Variables>,
+      FirebaseError,
+      FlattenedQueryResult<Data, Variables>,
+      QueryKey
+    >,
+    "queryFn" | "queryKey"
+  > {
+  queryFn: () => QueryRef<Data, Variables>;
+}
+
+/**
+ * injectDataConnectQuery takes a query ref and returns a wrapper function around Tanstack's `injectQuery`
+ * @param queryRefOrOptionsFn Query Ref or callback function for calling a new query
+ * @returns {CreateQueryResult<FlattenedQueryResult<Data, Variables>>}
+ */
+export function injectDataConnectQuery<Data, Variables>(
+  queryRefOrOptionsFn:
+    | QueryRef<Data, Variables>
+    | (() => CreateDataConnectQueryOptions<Data, Variables>),
+  injector?: Injector,
+  _callerSdkType: CallerSdkType = CallerSdkTypeEnum.TanstackAngularCore
+): CreateQueryResult<FlattenedQueryResult<Data, Variables>, FirebaseError> {
+  const finalInjector = injector || inject(Injector);
```

Comment:
I recommend using [EnvironmentInjector](https://angular.dev/api/core/EnvironmentInjector), here and elsewhere instead